proposals/devcontainer-templates.md

```diff
@@ -0,0 +1,192 @@
+| `description` | string | Description of the Template. |
+| `documentationURL` | string | Url that points to the documentation of the Template. |
+| `licenseURL` | string | Url that points to the license of the Template. |
+| `type` | string | Type of the dev container (image, dockerfile, dockerCompose) created by the Template. |
+| `options` | object | A map of options that the supporting tools should use to populate different configuration options for the Template. |
+| `platforms` | object | Languages and platforms supported by the Template. |
+| `keywords` | array | List of strings relevant to a user that would search for this Template. |
+
+### The `options` property
+
+The `options` property contains a map of option IDs and their related configuration settings. These `options` are used by the supporting tools to prompt the user to choose from different Template configuration options. The tools would replace the option ID with the selected value in the specified `replaceIn` files. This replacement would happen before dropping the `.devcontainer.json` (or `.devcontainer/devcontainer.json`) and other files (within the sub-directory of the Template) required to containerize your project. See [option resolution](#option-resolution) for more details. For example:
+
+```json
+{
+  "options": {
+    "optionId": {
+      "type": "string",
+      "description": "Description of the option",
+      "proposals": ["value1", "value2"],
+      "default": "value1",
+      "replaceIn": [".devcontainer.json"],
+    }
+  }
+}
+```
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| `optionId` | string | ID of the option used by the supporting tools to replace the selected value in the specified `replaceIn` files. |
+| `optionId.type` | string | Type of the option. Valid types are currently: `boolean`, `string` |
+| `optionId.description` | string | Description for the option. |
+| `optionId.proposals` | array | A list of suggested string values. Free-form values **are** allowed. Omit when using `optionId.enum`. |
+| `optionId.enum` | array | A strict list of allowed string values. Free-form values are **not** allowed. Omit when using `optionId.proposals`. |
+| `optionId.default` | string | Default value for the option. |
+| `optionId.replaceIn` | array | List of file paths which the supporting tool should use to perform the string replacement of `optionId` with the selected value. The provided path is always relative to the sub-directory of the Template. |
+
+> `Note`: The `options` must be unique for every `devcontainer-template.json`
+
+### Template Supporting Tools and Services
+
+Outlines tools and services that currently support adding Templates to a project.
+
+- [Visual Studio Code Remote - Containers](../docs/specs/supporting-tools.md/#visual-studio-code-remote---containers)
+- [GitHub Codespaces](../docs/specs/supporting-tools.md/#github-codespaces)
+
+## Adding a Template to a project or codespace
+  
```

Comment:
Since this section is tool / service specific, I'm not sure it belongs in the spec as much as it does documentation for VS Code and Codespaces. It could be added to tool specific steps.  A CLI-based example would be better if there's a need to have something here.

Reply:
Makes sense, removed!